spec/rubocop/cop/style/unneeded_condition_spec.rb

```diff
@@ -161,6 +173,19 @@
           ary << (foo || bar)
         RUBY
       end
+
+      it 'when `if` condition and `then` branch are the same ' \
+         'and it has no `else` branch' do
+        new_source = autocorrect_source(<<-RUBY.strip_indent)
+          if do_something
+            do_something
+          end
+        RUBY
+
+        expect(new_source).to eq(<<-RUBY.strip_indent)
+          do_something
```

Comment:
If `do_something` returns a truthy value, it will now be called once instead of twice. If it has side effects, that means changed semantics.